t/conftest.py

```diff
@@ -31,6 +39,117 @@ def app(celery_app):
+    finally:
+        for name in names:
+            try:
+                sys.modules[name] = prev[name]
+            except KeyError:
+                try:
+                    del sys.modules[name]
+                except KeyError:
+                    pass
+
+
+class _patching:
+
+    def __init__(self, monkeypatch, request):
+        self.monkeypatch = monkeypatch
+        self.request = request
+
+    def __getattr__(self, name):
+        return getattr(self.monkeypatch, name)
+
+    def __call__(self, path, value=SENTINEL, name=None,
+                 new=MagicMock, **kwargs):
+        value = self._value_or_mock(value, new, name, path, **kwargs)
+        self.monkeypatch.setattr(path, value)
+        return value
+
+    def object(self, target, attribute, *args, **kwargs):
+        return _wrap_context(
+            patch.object(target, attribute, *args, **kwargs),
+            self.request)
+
+    def _value_or_mock(self, value, new, name, path, **kwargs):
+        if value is SENTINEL:
+            value = new(name=name or path.rpartition('.')[2])
+        for k, v in kwargs.items():
+            setattr(value, k, v)
+        return value
+
+    def setattr(self, target, name=SENTINEL, value=SENTINEL, **kwargs):
+        # alias to __call__ with the interface of pytest.monkeypatch.setattr
+        if value is SENTINEL:
+            value, name = name, None
+        return self(target, value, name=name)
+
+    def setitem(self, dic, name, value=SENTINEL, new=MagicMock, **kwargs):
+        # same as pytest.monkeypatch.setattr but default value is MagicMock
+        value = self._value_or_mock(value, new, name, dic, **kwargs)
+        self.monkeypatch.setitem(dic, name, value)
+        return value
+
+    def modules(self, *mods):
+        modules = []
+        for mod in mods:
+            mod = mod.split('.')
+            modules.extend(reversed([
+                '.'.join(mod[:-i] if i else mod) for i in range(len(mod))
+            ]))
+        modules = sorted(set(modules))
+        return _wrap_context(module_context_manager(*modules), self.request)
+
+
+def _wrap_context(context, request):
+    ret = context.__enter__()
+
+    def fin():
+        context.__exit__(*sys.exc_info())
+    request.addfinalizer(fin)
+    return ret
+
+
+@pytest.fixture()
+def patching(monkeypatch, request):
+    """Monkeypath.setattr shortcut.
+    Example:
+        .. code-block:: python
+        >>> def test_foo(patching):
+        >>>     # execv value here will be mock.MagicMock by default.
+        >>>     execv = patching('os.execv')
+        >>>     patching('sys.platform', 'darwin')  # set concrete value
+        >>>     patching.setenv('DJANGO_SETTINGS_MODULE', 'x.settings')
+        >>>     # val will be of type mock.MagicMock by default
+        >>>     val = patching.setitem('path.to.dict', 'KEY')
+    """
+    return _patching(monkeypatch, request)
+
+
 @pytest.fixture(autouse=True)
 def test_cases_shortcuts(request, app, patching):
```

Comment:
As far as I can tell, the `patching` fixture was only thing that was coming from the `case` dependency. I took the current implementation from Celery (found here https://github.com/celery/celery/pull/7077)

Not sure if duplicating these ~100 lines of code is the best way forward but that does the job

Reply:
we might be using that from celery instead of case if those are already available there

Reply:
or we can handle this in a separate PR

Reply:
The version from Celery is defined under the tests directory (`celery/t`), which not included in [the distributions](https://github.com/celery/celery/blob/64d750bb113310503018c3c43a04d333fc1d2859/setup.py#L139), so it's not available in the version we have installed... [I tried](https://github.com/celery/django-celery-results/pull/461) to install from the source distribution, which seems to have the `t` directory included, but couldn't get to import the fixture.

Reply:
So I have 2 approaches:

1. Extract the commit as is from this PR https://github.com/celery/django-celery-results/pull/462 - just works but fixture is duplicated with Celery
2. Use the fixture from Celery https://github.com/celery/django-celery-results/pull/461 - not working right now, so I assume it needs more work in Celery... Perhaps move the fixture from `t` to under `celery` to be included in the distribution, but I'm not sure about the side effects of doing so.

Approach 1 is more practical, and can unblock this PR immediately. Approach 2 is more purist but needs more work and guidance from one of the Celery maintainers.